src/lib/lib.jl

```diff
@@ -43,6 +43,32 @@ unwrap(x) = x
 
 @adjoint unwrap(x) = unwrap(x), Δ ->(accum_param(__context__, x, Δ); (Δ,))
 
+# map
+
+# TODO: Replace with more performant version. See https://github.com/JuliaLang/julia/issues/13942.
+unzip(zipped) = collect(zip(zipped...))
+
+function sum_named_tuple(tpl)
```

Comment:
You should be able to replace this with `reduce(accum, xs)` (`accum` is our generalised `+`).